File: retos-python/08_carrera_renos.py
# Resultado 4 ⭐
def drawRace(indices, length):
    race = []

    for i in range(len(indices)): 
        
        track = indices[i]
        tmpDraw = list('~' * (length))

        if(track != 0):
            if(track > 0): 
                tmpDraw[track] = 'r'
            else:
                tmpDraw[length + track] = 'r'
        

        spaces = (' ') * (len(indices) - (i + 1));
        race.append(spaces + ''.join(tmpDraw) + ' /' + i+1);


    return '\n'.join(race)

File: retos-ts/08_carrera_renos.ts
// Resultado 5 ⭐
function drawRace(indices: number[], length: number): string {
    const race: string[] = [];

    for (let i = 0; i < indices.length; i++) {
        const track: number = indices[i];
        const tmpDraw: string[] = ('~').repeat(length).split('');


        if(track !== 0){
            (track > 0) 
                ? tmpDraw[track] = 'r'
                : tmpDraw[length + track] = 'r'
        }

        const spaces: string = (' ').repeat(indices.length - (i + 1));
        race.push(spaces + tmpDraw.join('') + ` /${i+1}`);
    }

    return race.join('\n');
}